app/services/filtered_artwork_search_service.rb

```diff
@@ -28,5 +46,11 @@ def headers
         'X-XAPP-TOKEN' => Rails.application.config_for(:gravity)['xapp_token']
       }
     end
+
+    def validate_parameters!(params)
+      actual_params = params.symbolize_keys.keys
+      unrecognized_params = actual_params - ALLOWED_PARAMS
+      raise ServiceError, "Unrecognized parameters: #{unrecognized_params.inspect}" if unrecognized_params.any?
+    end
```

Comment:
Having the allowlist also means we can validate all the query params here. Makes the contract with Gravity nice and explicit.